wled00/FX_fcn.cpp

```diff
@@ -230,16 +231,28 @@ CRGBPalette16 &Segment::loadPalette(CRGBPalette16 &targetPalette, uint8_t pal) {
   switch (pal) {
     case 0: //default palette. Exceptions for specific effects above
       targetPalette = PartyColors_p; break;
-    case 1: //periodically replace palette with a random one. Doesn't work with multiple FastLED segments
-      if (millis() - _lastPaletteChange > 5000 /*+ ((uint32_t)(255-intensity))*100*/) {
+    case 1: {//periodically replace palette with a random one. Transition palette change in 500ms
+      uint32_t timeSinceLastChange = millis() - _lastPaletteChange;
+      if (timeSinceLastChange > 5000 /*+ ((uint32_t)(255-intensity))*100*/) {
+        prevRandomPalette = randomPalette;
         randomPalette = CRGBPalette16(
                         CHSV(random8(), random8(160, 255), random8(128, 255)),
                         CHSV(random8(), random8(160, 255), random8(128, 255)),
                         CHSV(random8(), random8(160, 255), random8(128, 255)),
                         CHSV(random8(), random8(160, 255), random8(128, 255)));
         _lastPaletteChange = millis();
+        timeSinceLastChange = 0;
       }
-      targetPalette = randomPalette; break;
+      if (timeSinceLastChange <= 500) {
```

Comment:
timesincelastchange > 0 && timesincelastchange < 500

Reply:
unsigned int is always > 0 
😄

Reply:
Not exactly ... unsigned is always >= 0, so the corner case of timesincelastchange == 0 could be possible.

Reply:
Would there be a case for basing the transition time and speed on strip.getTransition() ?  (Say 5x for palette during time, 1x for transition time, with a minimum of 5s, 500ms respectively)

Reply:
> Would there be a case for basing the transition time and speed on strip.getTransition() ? (Say 5x for palette duration time, 1x for transition time, with a minimum of 5s, 500ms respectively)

No IMO as effect transition can be 0. We do want *some* transition for random palette change.

Reply:
> Not exactly ... unsigned is always >= 0, so the corner case of timesincelastchange == 0 could be possible.

You got me. 😁 Still, if timesincelastchange==0 there is not palette blending happening. `for` loop takes care of that.
You do not need to do extra comparison.

Code optimisation. 😉

Reply:
Ok 👍 point for you. Then not need to change that.